stl/inc/coroutine

```diff
@@ -11,16 +11,19 @@
 
 #ifdef _RESUMABLE_FUNCTIONS_SUPPORTED
 #pragma message("The contents of <coroutine> are not available with /await.")
-#pragma message("Remove /await for standard coroutines or use <experimental/coroutine> for legacy /await support.")
+#pragma message("Remove /await or use /await:strict for standard coroutines.")
+#pragma message("Use <experimental/coroutine> for legacy /await support.")
 #else // ^^^ /await ^^^ / vvv no /await vvv
-#ifndef __cpp_lib_coroutine
-#pragma message("The contents of <coroutine> are available only with C++20 or later.")
-#else // ^^^ __cpp_lib_coroutine not defined / __cpp_lib_coroutine defined vvv
+#if !defined(__cpp_lib_coroutine) && !defined(_DOWNLEVEL_COROUTINES_SUPPORTED)

```

Comment:
I realized that we don't actually have any `/await:strict` test coverage, either on GitHub (which is using VS 2019 16.10 Preview 1 which emits `Command line warning D9002 : ignoring unknown option '/await:strict'` without an error) or in the MSVC-internal test harness (where you're implementing `/await:strict`). Either way, the test itself is guarded with `__cpp_lib_coroutine`, so it'll do nothing in C++14/17 mode regardless of whether `/await:strict` is being ignored on GitHub or provided internally:

https://github.com/microsoft/STL/blob/194efde4bdc7d3e73984e266b983aed334288f2a/tests/std/tests/P0912R5_coroutine/test.cpp#L5

After talking about this on Discord, we agreed on the following strategy:

1. Update `yvals_core.h` to centralize the "downlevel" logic in the definition of `__cpp_lib_coroutine`. (It'll need to be moved outside of the `_HAS_CXX20` block that it's in, so it can additionally be activated for `_DOWNLEVEL_COROUTINES_SUPPORTED`.) @CaseyCarter noted that it should be fine to use the Standard value for downlevel support, since the thing we have to work around (spaceship) isn't observable to 14/17 code.
2. Don't directly test `_DOWNLEVEL_COROUTINES_SUPPORTED` here in `<coroutine>`, but do keep the `/await:strict` rephrasings.
3. Below, where we need to guard `<compare>` usage, instead of testing `__cpp_lib_coroutine`, or testing `__cpp_lib_three_way_comparison` (which is guarded by `__cpp_lib_concepts` so it's inactive for EDG at the moment), directly test `#if _HAS_CXX20`. That will then provide spaceship for all compilers in 20 mode. (The definition of spaceship will continue to contain `#ifdef __cpp_lib_concepts` guarding `compare_three_way{}`.)